src/bridge.js

```diff
@@ -12,10 +26,44 @@ export default class Bridge {
     this.alpine = reference
   }
 
-  init () {
-    // Once Turbolinks finished is magic, we initialise Alpine on the new page
-    // and resume the observer
+  // Tag cloaked elements so we can cloak them again before caching
+  tagCloakedElements (node) {
+    node.querySelectorAll('[x-cloak]').forEach((el) => {
+      el.setAttribute('data-alpine-was-cloaked', '')
+    })
+  }
+
+  removeAlpineGeneratedElements (node) {
```

Comment:
These two functions just wrap up the previous event handlers, pulled them into functions so it's a bit more obvious that a few unrelated things are happening in the event handlers.

Reply:
Cool, I might unify them. I know they are unrelated but querySelectorAll is a slow operation (it gets more obvious when your page is really big) so calling it multiple times has a price. I'll sleep on it and decide tomorrow/over the weekend. PR looks good, thanks for your help.